### PR TITLE
Drop tables / rows without required columns 

### PR DESCRIPTION
**Pull request by olejandro**
This PR drops tables / rows that either don't include required columns (i.e. tables) or values specified in required columns (rows).

### CODE REVIEW

File: xl2times/transforms.py [hunk 1]
```diff
@@ -222,6 +222,44 @@ def discard(table):
...
+    config: datatypes.Config,
+    tables: List[datatypes.EmbeddedXlTable],
+    model: datatypes.TimesModel,
+) -> List[datatypes.EmbeddedXlTable]:
+    """
+    Perform further validation of input tables by checking whether required columns are
+    present / non-empty. Remove tables without required columns or if they are empty.
+    """
+    result = []
+    for table in tables:
```
Comment (SamRWest) [inline xl2times/transforms.py:235]: This additional validation didn't fix my missing "uc_n" missing exception.

What _did_ work for austimes was to just drop empty dataframes, i.e. insert this block here
```python
        # drop empty dataframes
        if table.dataframe.empty:
            logger.warning(f"Dropping empty table {table.tag} in {table.filename}")
            continue
```

Comment (olejandro) [reply xl2times/transforms.py:235]: 🤣 That's a fun one! We actually drop all empty dataframes in `validate_input_tables`, but I guess at that point it is not yet empty. It probably has a comment row(s) that is dropped in `remove_comment_rows` afterwards.

Comment (SamRWest) [reply xl2times/transforms.py:235]: Yep, spot on - looks like that's exactly what is happening - there are tables where all the rows are commented (`\I:`) out.

Comment (olejandro) [reply xl2times/transforms.py:235]: Opened #201 to handle this. @SamRWest any chance you could test?

File: xl2times/transforms.py [hunk 2]
```diff
@@ -222,6 +222,44 @@ def discard(table):
...
+        tag = datatypes.Tag(table.tag.split(":")[0])
+        required_cols = config.required_columns[tag]
+        unique_table_cols = set(table.dataframe.columns)
+        if required_cols:
+            # Drop table if any column in required columns is missing
+            if not required_cols.issubset(unique_table_cols):
+                missing_cols = required_cols - unique_table_cols
```
Comment (siddharth-krishna) [inline xl2times/transforms.py:242]:
```suggestion
            missing_cols = required_cols - unique_table_cols
            if missing_cols:
```
Avoids recomputing the set difference :)

**Review by siddharth-krishna — APPROVED** [review]
LGTM thanks. Let some minor suggestions on improving the code.

**Pull request merged**